src/wallet/bdk.rs

```diff
@@ -1363,6 +1451,142 @@ impl Wallet {
     }
 }
 
+//___________________WALLET/DESCRIPTOR______________________________________________________________
+
+/// Wrapper for `IntoWalletDescriptor` that performs additional checks on the keys contained in the
+/// descriptor
+pub(crate) fn into_wallet_descriptor_checked<T: IntoWalletDescriptor>(
+    inner: T,
+    secp: &Secp256k1<All>,
+    network: Network,
+) -> Result<(ExtendedDescriptor, KeyMap), DescriptorError> {
```

Comment:
I love generics, but when we are creating our specialized functions for specific tasks, use explicit data types. This makes it easy for onlookers to understand what's happening without searching for trait definitions. For ex: currently, you are just passing a string as inner. So just type it as a String.

Also this should not be a public API. it's pretty inside the wallet. Generics make more sense in public APIs.